Dynatrace/dynatrace-operator
# Moves deploymentmetadata to configmap

# Description

Currently(previous this PR) the deployment-metadata is set in the following ways:
- as a set of arguments for the oneagent daemonset
	- this can cause an restart if the metadata changes, which should not happen
	- different way to the same thing (using args instead of envvars) for no reason (probably because of this part of the logic is quite old)
- as a static envvar for the activegate statefulset
   - this can cause an restart if the metadata changes, which should not happen
- as a static envvar for the injected user pods
   - this is fine

This changes it to the following:
1. We create a `configMap` for each `dynakube` with a name `<name-of-dynakube>-deployment-metadata`.
   - This `configMap` has 2 keys, `oneagent` and `activegate`, each containing a comma separated strings of deployment metadata
   - If only ActiveGate is configured then only the `activegate` key is present same for the OneAgent
2. We provide the deployment-metadata with an environment variable in all cases.
3. For the ActivGate statefulSet and OneAgent daemonSet the deployment-metadata envvar references the corresponding key in the `<name-of-dynakube>-deployment-metadata` configMap

Some misc:
Some consts were moved around to make more sense.
(using a const from the activegate package in the oneagent package also the other way around is a bit wrong)

## How can this be tested?
Deploy a dynakube, configured with OneAgent and/or ActiveGate.
Check if the configmap is present and is used by the statefulSet and daemonSet


## Checklist
